app/models/chess_piece.rb

```diff
@@ -22,6 +24,20 @@ def horizontally_obstructed?(endpoint_x, endpoint_y)
     end
   end
 
+  def diagonally_obstructed?(endpoint_x, endpoint_y)
+    if endpoint_x > position_x
+      ((position_x + 1)..endpoint_x).find do |x|
+        position_y = x 
```

Comment:
you're going to have to use a different variable name.

Because we have a column on `ChessPiece` called `position_y`, `Rails` will create an `attr_accessor` for the column. I.e. `ChessPiece` has methods: `position_y` and `position_y=`. in this case you're actually calling the latter method and assigning `x` to the record's `position_y` attribute =(